Formula/c-kermit.rb

```diff
@@ -61,3 +69,339 @@ def install
      x = (int) ((stdin->_IO_read_end) - (stdin->_IO_read_ptr));
      debug(F101,"cmdconchk _IO_file_flags","",x);
  #else  /* _IO_file_flags */
+diff --git a/ck_ssl.c b/ck_ssl.c
```

Comment:
I'm a bit skeptical about applying this patch since it's not clear that this has been reviewed by anyone as-is.

However, if you can get this hosted somewhere official-ish with respect to the Kermit Project, plus a signoff from someone else upstream that this is a reasonable patch, then I think we can consider applying it.

Reply:
@carlocab I've asked the maintainer of CKW (the Windows port of C-Kermit) to review and test the change ... hopefully he'll have time to check it out

Reply:
I'm not so much concerned about whether the patch does what you say it does -- I'm more concerned about this patch unintentionally introducing security vulnerabilities to this software.

That's not a determination I can make for myself, which is why I'd rather someone who were familiar with the codebase to review the patch. Hosting it somewhere official also helps make others more confident that this patch was reviewed by upstream, and makes it more useful to other users who don't use Homebrew.

Reply:
Understood.

Currently, Kermit is maintained by just one person, Frank Da Cruz (and he also isn't the one maintaining the SSL support code).  David (the one I've requested do the review, when he has time) has been active in updating the OpenSSL support as part of the CKW branch, so should be able to alleviate any concerns.

There really is no longer any upstream support of the old versions, and they're not going to be fixed - I'll try to explain a bit of about the Kermit Project ... C-Kermit 10 is very likely going to be the final "Kermit Project" release of C-Kermit, due to the cancellation of the project and it's funding (including the layoff of all staff involved) at Columbia. The upstream (just Frank, and doing the work on his own) has been working to get a new "10.0" release out, and then move the project under community control.

The cancellation of the upstream project was announced on March 15th, 2011 and set be finalized in July 2011.

As part of the cancellation, certain materials such as the documentation became fully open source licensed, and a "final" version of C-Kermit was wrapped up (9.0.300), and released on June 28th, 2011.  There were two small point releases, and then came the final cancellation of the Kermit Project, so, there have been no further "stable" versions of Kermit since July 2011.

In the twelve years after the projects cancellation, the original author (Frank) has continued to release updates and provide support on his own.  Frank doesn't give these releases "dot zero" versions, so, unfortunately, that kind of clashes with your policy of using "releases", so we're stuck using the version that came out almost 13 years ago.  All the current work is going to 10.0 release, which should be the first fully "community" C-Kermit release version rather than a product of the university.

Other distributions that ship TLS-enabled Kermit (like FreeBSD, Debian, etc) are not shipping this old release version - they're using later versions.  There are lots of bugs but I've not made any attempt to fix them - they are all already fixed upstream.  I've done the smallest amount of work possible to just change the code to compile and link with current OpenSSL.

Hopefully David has time to review the patch soon.

Reply:
Hi @carlocab, I'm the maintainer of the [Windows port](https://github.com/davidrg/ckwin) and have dealt with this code in the last few months adding in support for newer OpenSSL versions along with the TLS SNI extension. The patch looks fine to me - the end result is much the same as what's now in recent C-Kermit 10 betas, just without some old cruft to support building with OpenSSL 0.9.8 (_long_ out of support now) and some minor updates to support OpenSSL 3.x and TLS SNI.

Reply:
Thanks - I can update this to use OpenSSL 3 if that is preferred?  Seems 1.1 is EOL soonish. 

(Supporting SNI would be easy too, but since it would be adding a new feature, I didn't do it as I don't believe feature backports are appropriate for Homebrew.)

Reply:
If other package managers are doing so as well I'd rather have a beta then a patch that cumulates a decade of bugfixes.

Reply:
Let me test a version bump and I'll get back to you on it ...